build-tools/packages/build-cli/src/library/typescriptApi.ts

```diff
@@ -42,49 +44,73 @@ function isTypeExport(_decl: ExportedDeclarations): boolean {
 }
 
 /**
- * Searches given JSDocs for known {@link ApiTag} tag.
+ * Searches given JSDocs for known {@link ApiTag} tags.
  *
- * @returns Recognized {@link ApiTag} from JSDocs or undefined.
+ * @returns Recognized {@link ApiTag}s from JSDocs or undefined.
  */
-function getApiTagFromDocs(jsdocs: JSDoc[]): ApiTag | undefined {
+function getApiTagsFromDocs(jsdocs: JSDoc[]): ApiTag[] | undefined {
+	const tags: ApiTag[] = [];
 	for (const jsdoc of jsdocs) {
 		for (const tag of jsdoc.getTags()) {
 			const tagName = tag.getTagName();
 			if (isKnownApiTag(tagName)) {
-				return tagName;
+				tags.push(tagName);
 			}
 		}
 	}
-	return undefined;
+	return tags.length > 0 ? tags : undefined;
 }
 
 /**
- * Searches given Node's JSDocs for known {@link ApiTag} tag.
+ * Searches given Node's JSDocs for known {@link ApiTag} tags.
  *
- * @returns Recognized {@link ApiTag} from JSDocs or undefined.
+ * @returns Recognized {@link ApiTag}s from JSDocs or undefined.
  */
-function getNodeApiTag(node: Node): ApiTag | undefined {
+function getNodeApiTags(node: Node): ApiTag[] | undefined {
 	if (Node.isJSDocable(node)) {
-		return getApiTagFromDocs(node.getJsDocs());
+		return getApiTagsFromDocs(node.getJsDocs());
 	}
 
 	// Some nodes like `ExportSpecifier` are not JSDocable per ts-morph, but
 	// a JSDoc is present.
 	const jsdocChildren = node.getChildrenOfKind(SyntaxKind.JSDoc);
 	if (jsdocChildren.length > 0) {
-		return getApiTagFromDocs(jsdocChildren);
+		return getApiTagsFromDocs(jsdocChildren);
 	}
 
 	// Some nodes like `VariableDeclaration`s are not JSDocable, but an ancestor
 	// like `VariableStatement` is and may contain tag.
 	const parent = node.getParent();
 	if (parent !== undefined) {
-		return getNodeApiTag(parent);
+		return getNodeApiTags(parent);
 	}
 
 	return undefined;
 }
 
+/**
+ * Searches given Node's JSDocs for known {@link ApiTag} tags and derive export level.
```

Comment:
Probably a good place to describe the interaction between `@legacy` and the standard release tags.